### DISCUSSION TITLE
Is there a way to extract query args from a SQLiteException?

### DISCUSSION THREAD

**Discussion by MaxiStefan** [simolus3/drift#2338] [Q&A] [answered]
Hi, 

I want to log what goes wrong (if anything does) with database queries and so far if I try to cause exceptions the causing statements only prints the executed query and it does not show the values that go with it as the database logs do. Is there any way to achieve what I need? See screenshots below.
![image](https://user-images.githubusercontent.com/48346391/223114749-4e0da1ca-9156-4e6b-b5c4-18fe8fc6fb1b.png)
![image](https://user-images.githubusercontent.com/48346391/223114846-d6ab41bc-2788-4efc-9278-3a4bc9518fc7.png)

Thanks in advance

**Comment by simolus3** [comment, marked as answer]
Thanks for the request! This has actually been implemented already, it looks like I just forgot to publish. This feature should be available in version `1.9.2` of the `sqlite3` package, which adds the `parametersToStatement` field to the exception class.

**Reply by MaxiStefan** [reply]
Thanks! :)

**Reply by MaxiStefan** [reply]
Is there any extra configuration I need to do on my end? Such as drift build options?

I am running Drift 2.1.0 and sqlite3_flutter_libs 0.50

**Reply by kuhnroyal** [reply]
Probably run `pub upgrade sqlite3`. But I noticed that it still doesn't work for non-prepared statements and created https://github.com/simolus3/drift/issues/2340 because I didn't see this discussion before.

**Reply by simolus3** [reply]
Yeah, generally the values of parameters should be available (I've added tests for this to the `sqlite3` package and now in drift as well).

Are you catching a `Sqlite3Exception` directly? When using `NativeDatabase.createInBackground` or custom drift isolates, you'd only be getting a `DriftRemoteException` (although that will also improve). If you do get an exception that doesn't have the parameters set, do you have a way to reproduce that?
